cmd/libvirt-actuator/main.go

```diff
@@ -109,7 +133,244 @@ func init() {
+
+func cmdRun(binaryPath string, args ...string) ([]byte, error) {
+	cmd := exec.Command(binaryPath, args...)
+	return cmd.CombinedOutput()
+}
+
+func bootstrapCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bootstrap",
+		Short: "Bootstrap kubernetes cluster with kubeadm",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			libvirturi := cmd.Flag("libvirt-uri").Value.String()
+			if libvirturi == "" {
+				return fmt.Errorf("--libvirt-uri needs to be set")
+			}
+
+			inclusterlibvirturi := cmd.Flag("in-cluster-libvirt-uri").Value.String()
+			if inclusterlibvirturi == "" {
+				return fmt.Errorf("--in-cluster-libvirt-uri needs to be set")
+			}
+
+			libvirtpk := cmd.Flag("libvirt-private-key").Value.String()
+			if libvirtpk == "" {
+				return fmt.Errorf("--libvirt-private-key needs to be set")
+			}
+
+			masterguestpk := cmd.Flag("master-guest-private-key").Value.String()
+			if masterguestpk == "" {
+				return fmt.Errorf("--master-guest-private-key needs to be set")
+			}
+
+			testNamespace := &apiv1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+			}
+
+			glog.Infof("Creating secret with the libvirt PK from %v", libvirtpk)
+			libvirtPKSecret, err := BuildPKSecret("libvirt-private-key", testNamespace.Name, libvirtpk)
+			if err != nil {
+				return fmt.Errorf("unable to create libvirt-private-key secret: %v", err)
+			}
+
+			machinePrefix := cmd.Flag("environment-id").Value.String()
+
+			testCluster := &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      machinePrefix,
+					Namespace: testNamespace.Name,
+				},
+				Spec: clusterv1.ClusterSpec{
+					ClusterNetwork: clusterv1.ClusterNetworkingConfig{
+						Services: clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.0.0.1/24"},
+						},
+						Pods: clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.0.0.1/24"},
+						},
+						ServiceDomain: "example.com",
+					},
+				},
+			}
+
+			// Create master machine and verify the master node is ready
+			masterUserDataSecret, err := manifests.MasterMachineUserDataSecret(
+				"masteruserdatasecret",
+				testNamespace.Name,
+				[]string{"127.0.0.1"},
+			)
+			if err != nil {
+				return err
+			}
+
+			masterMachineProviderConfig, err := testutils.MasterMachineProviderConfig(masterUserDataSecret.Name, libvirturi)
+			if err != nil {
+				return err
+			}
+			masterMachine := manifests.MasterMachine(testCluster.Name, testCluster.Namespace, masterMachineProviderConfig)
+
+			glog.Infof("Creating master machine")
+
+			actuator := utils.CreateActuator(masterMachine, masterUserDataSecret)
+			err = actuator.Create(testCluster, masterMachine)
+			if err != nil {
+				return err
+			}
+
+			lcw, err := libvirtclient.NewLibvirtClient(libvirturi)
+			if err != nil {
+				return err
+			}
+
+			// Wait until the instance has the ip address
+			var masterMachinePrivateIP string
+			err = wait.Poll(pollInterval, timeoutPoolInterval, func() (bool, error) {
```

Comment:
We don't appear to check the state of this error value.

Reply:
Done